httpx/dispatch/http2.py

```diff
@@ -58,10 +58,10 @@ def __init__(
     async def close(self) -> None:
         await self.stream.close()
 
-    def initiate_connection(self) -> None:
+    async def initiate_connection(self, timeout: TimeoutConfig = None) -> None:
```

Comment:
What do you think of instead of making this function `async` we keep this sync but just don't write anything? We're gonna get a `await stream.write()` within `send_headers()` anyways?

Reply:
True that, but wouldn't it become misleading to send data not related to headers in `send_headers()`? (For example, `initiate_connection()` [sends the HTTP/2 settings frame](https://python-hyper.org/projects/h2/en/stable/_modules/h2/connection.html#H2Connection.initiate_connection) in there.)

If we really want to keep `write_no_block()` (which, I start to realize, perhaps initially existed because we'd like to send headers along with the initial frame in one go, instead of accessing the socket twice?), I can look at implementing a simple buffer mechanism in the trio backend.

Reply:
> I can look at implementing a simple buffer mechanism in the trio backend.

I did just that — turns out to be very simple, and requiring no changes to the concurrency backend API.